### DISCUSSION TITLE
Show InfoBar without BuildContext

### DISCUSSION THREAD

**Discussion by Nilsuuus** [bdlukaa/fluent_ui#989] [Q&A] [answered]
Is there a way to show an InfoBar without having the BuildContext?
MaterialApp has a parameter called scaffoldMessengerKey which allows you to show a Snackbar without having direct access to a BuildContext. So if you have any operation which is far away from the widget tree and the result needs to be delivered to the user, this is very usefull.

Is there a way to get it done if you use FluentApp instead of MaterialApp?

**Comment by bdlukaa** [comment, marked as answer]
¹ You not having access to a build context is probably a design fault in your app.
² Yes, it is totally possible! You can use a `navigatorKey` to access the navigator context:

```dart
final navigatorKey = GlobalKey<NavigatorState>();

FluentApp(
  navigatorKey: navigatorKey,
),

final context = navigatorKey.currentContext;
if (context != null) {
  ...
}
```
